api/api.ts

```diff
@@ -34,22 +38,10 @@ app.post('/upload/:tenant', async (req: express.Request, res: express.Response)
             throw new Error('Error fetching Authentication API')
         }
 
-        let data: Buffer[] = []
         const filename: string = getFilename(req)
-
-        req.on('data', (chunk: Buffer) => data.push(chunk))
-
-        req.on('end', () => {
-            fs.writeFile(`./uploads/${filename}`, Buffer.concat(data), (err: NodeJS.ErrnoException | null) => {
-                if (err) {
-                    console.error(err)
-                    res.status(500).send(`Error saving file ${filename}`)
-                } else {
-                    console.log(`File ${filename} saved`)
-                    res.status(200).send(`File ${filename} saved.`)
-                }
-            })
-        })
+        const tenantConfig = await getDocByTenant(req.params.tenant)
+        const uploadJsonResponse = await uploadFile(tenantConfig, req.body, filename)
```

Comment:
`uploadJsonResponse` never used, even if a fail would happen in the upload the `File ${filename} saved` message would be returned.

Reply:
Avoid guiding application to be exception driven. Not a great way to control flow.

It is easy to mess. Prefer doing stuff explicitly. Use a tagged union to deal with multiple possible returns, if it is the case.

Like:

```ts
type Maybe = Err | Success;
type Success = {
  status: “ok”,
  resultField1: typeField1,
  resultField2: typeField2,
    . . .
};
type Err = {
  status: “fail”,
  reason: string,
  errField1: typeField1,
  errField2: typeField2,
    . . .
};
```

Reply:
But this way, I have to check if msg can be send, right?
```ts
if (typeof  uploadJsonResponse === "Success") {
    res.send("file saved successfully")
}
```